### PR TITLE
Chapter 4 5

### PR DESCRIPTION
**Pull request by puteriffah**

### CODE REVIEW

**Comment by puteriffah** [comment]
ready

File: config/application.rb [hunk 1]
```diff
@@ -10,4 +10,4 @@ class Application < Rails::Application
     config.i18n.available_locales = [:en, :ru]
     config.i18n.default_locale = :en
   end
-end
+end
```
Comment (ngocvt-0484) [inline config/application.rb]: shoud follow by number 12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

File: config/settings.yml [hunk 2]
```diff
@@ -3,3 +3,4 @@ links:
     home_page: https://www.railstutorial.org/
     help_page: http://www.railstutorial.org/help
     book_page: https://www.railstutorial.org/book
+
```
Comment (ngocvt-0484) [inline config/settings.yml]: remove empty line
and follow number 12 https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -10,7 +10,12 @@
   </head>
 
   <body>
-    <%= yield %>
+  <%= render "layouts/header"%>
+
+  <div class="container">
+  <%= yield %>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb]:
```suggestion
    <%= yield %>
```

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  <%= link_to "sample app", "#", id: "logo" %>
+  <nav>
+  <ul class="nav navbar-nav navbar-right">
+  <li><%= link_to "Home", "#" %></li>
+  <li><%= link_to "Help", "#" %></li>
+  <li><%= link_to "Log in", "#" %></li>
+  </ul>
+  </nav>
+  </div>
+</header>
```
Comment (ngocvt-0484) [inline app/views/layouts/_header.html.erb]: shoud follow by number 12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+  <%= link_to "sample app", "#", id: "logo" %>
+  <nav>
+  <ul class="nav navbar-nav navbar-right">
+  <li><%= link_to "Home", "#" %></li>
```
Comment (ngocvt-0484) [inline app/views/layouts/_header.html.erb]: you need tab 2 space to format html

File: app/assets/stylesheets/custom.scss [hunk 6]
```diff
@@ -0,0 +1,32 @@
+/* universal */
+body {
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/custom.scss:3]:
```suggestion
body {
  padding-top: 60px;
}

section {
  overflow: auto;
}

textarea {
  resize: vertical;
}

.center {
  text-align: center;
}

.center h1 {
  margin-bottom: 10px;
}```

Comment (ngocvt-0484) [reply app/assets/stylesheets/custom.scss:3]: change all

File: app/assets/stylesheets/custom.scss [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    color: #fff;
+    text-transform: uppercase;
+    letter-spacing: -1px;
+    padding-top: 9px;
+    font-weight: bold;
+    }
+    #logo:hover {
+    color: #fff;
+    text-decoration: none;
+    }
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/custom.scss]: shoud follow by number 12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

**Comment by ngocvt-0484** [comment]
please change 3 commits to 1 commit
you can use `git rebase -i HEAD~3`

**Comment by puteriffah** [comment]
ready

**Comment by vanvtt-0952** [comment]
commented
please change 3 commits to 1 commit
you can use git rebase -i HEAD~3

**Comment by vanvtt-0952** [comment]
plz, fix conflict code

**Comment by puteriffah** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 8]
```diff
@@ -0,0 +1,32 @@
+/* universal */
+body {
+    padding-top: 60px;
```
Comment (vanvtt-0952) [inline app/assets/stylesheets/custom.scss]: use 1 tab = 2 spaces, plz

Comment (puteriffah) [reply app/assets/stylesheets/custom.scss]: image
like this?

**Comment by puteriffah** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  margin-bottom: 10px;
+}
+
+/* header */
+#logo {
+  float: left;
+  margin-right: 10px;
+  font-size: 1.7em;
+  color: #fff;
+text-transform: uppercase;
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]:
```suggestion
  text-transform: uppercase;
```

File: app/assets/stylesheets/custom.scss [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+  letter-spacing: -1px;
+  padding-top: 9px;
+  font-weight: bold;
+}
+#logo:hover {
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]:
```suggestion
  font-weight: bold;
}

#logo:hover {
```

File: app/views/layouts/_header.html.erb [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    <div class="container">
+    <%= link_to "sample app", "#", id: "logo" %>
+    <nav>
+    <ul class="nav navbar-nav navbar-right">
+    <li><%= link_to "Home", "#" %></li>
+    <li><%= link_to "Help", "#" %></li>
+    <li><%= link_to "Log in", "#" %></li>
+    </ul>
+    </nav>
+    </div>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb]:
```suggestion
    <nav>
      <ul class="nav navbar-nav navbar-right">
        <li><%= link_to "Home", "#" %></li>
        <li><%= link_to "Help", "#" %></li>
        <li><%= link_to "Log in", "#" %></li>
      </ul>
    </nav>
```

**Comment by puteriffah** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+  font-size: 1.7em;
+  color: #fff;
+text-transform: uppercase;
+  letter-spacing: -1px;
+  padding-top: 9px;
+  font-weight: bold;
+}
+
+#logo:hover {
+  color: #fff;
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]: create color variable in other file, then call back here

**Comment by puteriffah** [comment]
ready

File: app/assets/stylesheets/color.scss [hunk 13]
```diff
@@ -0,0 +1,3 @@
+$light-gray: #777;
+$gray-medium-light: #eaeaea;
+$white: #fff;
```
Comment (saobv-2223) [inline app/assets/stylesheets/color.scss]: add an empty line after line 3, please

File: app/views/layouts/_header.html.erb [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+  <header class="navbar navbar-fixed-top navbar-inverse">
+    <div class="container">
+    <%= link_to "sample app", "#", id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home", "#" %></li>
+        <li><%= link_to "Help", "#" %></li>
+        <li><%= link_to "Log in", "#" %></li>
+      </ul>
+    </nav>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb]: indenting 2 spaces for these lines

**Comment by puteriffah** [comment]
ready

**Review by chungpv-1008 — COMMENTED** [review]
approved

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**